<a href="https://colab.research.google.com/github/ctruciosm/BasicStats/blob/main/Exercicio_ANOVA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Amostras aleatórias foram selecionadas de três populações com distribuições $N(\mu_1, \sigma)$, $N(\mu_2, \sigma)$ e $N(\mu_3, \sigma)$. Os dados obtidos são os seguintes: 

- **Amostra 1**: 93, 98, 107, 102
- **Amostra 2**: 77, 87, 84, 95, 85
- **Amostra 3**: 88, 75, 73, 84, 74


Com um nível de significância de 5\%, podemos rejeitar a Hipótese nula de que as médias das três populações são iguais?. Assuma que as variâncias dos três grupos é a mesma.

**Solução:**

- Passo 1: Definir as hipóteses apropriadamente.

$$H_0: \mu_1 = \mu_2 = \mu_3 \quad  \text{vs}  \quad H_1: H_0 \quad \text{não é verdade}$$

- Passo 2: Definimos como faremos o teste.

Neste caso, como queremos testar a igualdade de mais de 2 grupos, não podemos mais utilizar comparação de médias. Utilizaremos **ANOVA**

Para fazer ANOVA, um caminho "simples" é preencher a seguinte tabela:

| Fonte de Variação | g.l  |  Soma dos Q.  | Q. Médios | F     |
|:-----------------|:----|:--------------------:|:-----------------:|:----:|
| Tratamento        |  $k-1$ |  SQTr                | $QMTr = \dfrac{SQTr}{k-1}$ |  $\dfrac{QMTr}{QME}$ |
| Erro              | $n_T - k$ |  SQE              | $QME = \dfrac{SQE}{n_T - k}$ |        |
| Total             | $n_T-1$   | SQT               |                       |        |



em que:

- $n_T$ é o número total de observações (somados todos os grupos).
- $k$ é o número de grupos (no nosso caso $k = 3$)
- $SQTr = \displaystyle \sum_{j=1}^{k} n_j (\bar{x}_{j} - \bar{\bar{x}})^2$
- $SQE = \displaystyle \sum_{j = 1}^k \sum_{i=1}^{n_j}(x_{ij} - \bar{x}_j)^2$
- $\bar{\bar{x}}$ é a média de todas as observações


In [8]:
# Preenchemos as observações
amostra1 = c(93, 98, 107, 102)
amostra2 = c(77, 87, 84, 95, 85)
amostra3 = c(88, 75, 73, 84, 74)
amostra_global = c(amostra1, amostra2, amostra3) # todos os elementos

# Calculamos as médias:
m_g = mean(amostra_global)  # média global
m_1 = mean(amostra1) # média da amostra 1
m_2 = mean(amostra2) # média da amostra 2
m_3 = mean(amostra3) # média da amostra 3

# Tamanhos de amostra em cada grupo
n1 = length(amostra1) # Obs na amostra 1
n2 = length(amostra2) # Obs na amostra 2
n3 = length(amostra3) # Obs na amostra 3
nT = n1 + n2 + n3     # Total de Obs (juntando todos os grupos)



In [11]:
# Imprimimos os ns apenas para ver os valores
n1
n2
n3
nT

[1] 4

[1] 5

[1] 5

[1] 14

In [10]:
# Calculamos as somas de quadrados para preencher a tabela

# Soma de Quadrados Totais
SQT = sum((amostra_global - m_g)^2) 
# Soma de Quadrados dos Tratamentos
SQTr = n1*(m_1-m_g)^2 + n2*(m_2-m_g)^2 + n3*(m_3-m_g)^2
# Soma de Quadrados dos Erros
SQE = SQT - SQTr
# Imprimindo resultados
c(SQT, SQTr, SQE)

[1] 1476.857 1020.857  456.000

Preenchemos a tabela


| Fonte de Variação | g.l  |  Soma dos Q.  | Q. Médios | F     |
|:-----------------|:----|:--------------------:|:-----------------:|:----:|
| Tratamento        |  2 |   1020.86             | 1020.86/2 | (1020.86/2 )/4567/11 |
| Erro              | 11 |   456         | 4567/11 |        |
| Total             | 12  | `             |                       |        |

In [14]:
(1020.86/2 )/4567/11

[1] 0.01016044

In [ ]:
# Rejeitamos H0 se F > 

- Passo 3: Definir a regra de decisão:

Rejeitamos $H_0$ se $F > F (1-\alpha, k − 1, nT − k)$. Então precisamos saber qual o valor de $F (1-\alpha, k − 1, nT − k)$

In [15]:
alpha = 0.05
qf(1-alpha, 3 - 1, 14 - 3)

[1] 3.982298

0.0101604395166909 > 3.98229795709448? Não, então não rejeitamos $H_0$